staging/com.smartnews/index.js

```diff
@@ -0,0 +1,73 @@
+"use strict";
+
+const Tp = require("thingpedia");
+const API_URL = "https://039ev0y88l.execute-api.us-west-1.amazonaws.com/test/v2";
+const USER = "jack";  // use 1 for now
+//const cheerio = require("cheerio");
+
+module.exports = class SmartNewsDevice extends Tp.BaseDevice {
+    constructor(engine, state) {
+         super(engine, state);
+
+         this.uniqueId = "com.smartnews";
+         this.name = "SmartNews";
+         this.description = "SmartNews latest articles";
+    }
+
+    //connect SN API endpoint /top with GET request
+    get_top_articles({counter}) {
+        counter = counter || 10; //default is 10 news
+        let url = API_URL + "/top?counter=" + counter;
+        return Tp.Helpers.Http.get(url).then((response) => {
+            return JSON.parse(response);
+        }).then((parsed) => {
+            return parsed.map((article) => {
+                return {
+                    news_id: article["id"],
+                    title: article["title"],
+                    date: new Date(article["published_time"] * 1000),
+                    site_name: article["site"],
+                    audio: article["audio"],
+                    content: article["content"],
+                    url: article["url"]
+                };
+            });
+        });
+    }
+    //connect SN API endpoint /list with GET request
+    get_reading_list({user=USER}) {
```

Comment:
`user` is not an input parameter to this function. Instead, if you want per-user functionality you must change the configuration mechanism of the device, set it to OAuth, and use that information to access the user ID / username.